python/eos/pyhf.py

```diff
@@ -0,0 +1,85 @@
+        self.parameter_map = parameter_map
+
+        if isinstance(workspace, pyhf.workspace.Workspace):
+            self.workspace = workspace
+        else:
+            try:
+                with open(workspace, 'r') as f:
+                    f = json.load(f)
+                self.workspace = pyhf.Workspace(f)
+            except:
+                raise ValueError('`workspace` must be a pyhf workspace or json file specifying one. ')
+
+        self.model = self.workspace.model()
+        self.data = np.array(self.workspace.data(self.model, include_auxdata=False))
+
+        self.varied_parameters = []
+        for pn, v in zip(self.model.config.par_names, self.model.config.suggested_init()):
+            if pn in parameter_map:
+                qn = eos.QualifiedName(parameter_map[pn])
+            else:
+                qn = eos.QualifiedName('pyhf::' + pn)
+
+            if self.parameters.had(qn):
+                p = self.parameters[qn]
+            else:
+                p = self.parameters.declare(qn, v)
+
+            p.set(v)
+            self.varied_parameters.append(p)
+            
+            o = eos.Observable.make(qn, self.parameters, eos.Kinematics(), eos.Options())
+            self.cache.add(o)
+
+        self.number_of_observations = int(1)
+
+    def evaluate(self):
+        parameter_values = np.array([p.evaluate() for p in self.varied_parameters])
+        # only main term in pyhf - constraints are handled in EOS
+        return self.model.mainlogpdf(self.data, parameter_values).item()
+
+    @staticmethod
+    def factory(cache, workspace, parameter_map={}):
+        return PyhfLogLikelihood(cache, workspace, parameter_map)
+
+    def priors(self):
+        priors = []
+        for name, _ in self.model.config.modifiers:
+            par_name = self.parameter_map[name] if name in self.parameter_map else 'pyhf::' + name
+            param_set = self.model.config.param_set(name)
+            
+            if not param_set.constrained:
+                prior = {
+                    'parameter': par_name,
+                    'min': param_set.suggested_bounds[0][0],
+                    'max': param_set.suggested_bounds[0][1],
+                    'type': 'uniform',
+                }
+                priors.append(prior)
+                continue
+
+            if param_set.pdf_type == 'normal':
+                for i in range(param_set.n_parameters):
+                    prior = {
+                        'parameter': par_name + ('' if param_set.is_scalar else f'[{i}]'),
+                        'min': param_set.suggested_bounds[i][0], 
+                        'max': param_set.suggested_bounds[i][1],
+                        'type': 'gauss',
+                        'central': param_set.auxdata[i],
+                        'sigma': param_set.width()[i]
+                    }
+                    priors.append(prior)
+
+            elif param_set.pdf_type == 'poisson':
+                raise NotImplementedError
```

Comment:
This can now be implemented since you added ``eos::priors::Poisson``, right?

Reply:
My previous comment still stands. I had already removed the ``min``/``max`` values to use a "true" Gaussian prior w/ infinite support, rather than the curtailed Gaussian we were limited to in past EOS versions.

Reply:
We just fixed that. Resolved.